File: Backend/src/Sofco.Model/Models/Admin/Module.cs
using System.Collections.Generic;
using Sofco.Model.Interfaces;
using Sofco.Model.Relationships;

namespace Sofco.Model.Models.Admin
{
    public class Module : BaseEntity, ILogicalDelete
    {
        public string Description { get; set; }
        public bool Active { get; set; }
        public string Code { get; set; }

        public Menu Menu { get; set; }

        public IList<RoleModule> RoleModule { get; set; }
        public IList<ModuleFunctionality> ModuleFunctionality { get; set; }
    }
}

File: Frontend/src/app/models/admin/module.ts
import { Menu } from "app/models/admin/menu";
import { Functionality } from "app/models/admin/functionality";

export class Module {
    constructor(
        public id: number,
        public code: string,
        public description: string,
        public active: boolean,
        public menuId: number,
        public menu: Menu,
        public functionalities: Functionality[] = null,
        public available: boolean = false
    ){}
}